pynicotine/chatrooms.py

```diff
@@ -79,19 +84,18 @@ def remove_room(self, room):
 
         if room == "Public ":
             self.queue.append(slskmessages.LeavePublicRoom())
-        else:
-            self.queue.append(slskmessages.LeaveRoom(room))
 
-        self.joined_rooms.discard(room)
+        elif room in self.joined_rooms:
+            log.add_chat(_(f"Leaving room {room}"))
+            self.queue.append(slskmessages.LeaveRoom(room))
+            self.joined_rooms.discard(room)
 
         if room in self.config.sections["columns"]["chat_room"]:
             del self.config.sections["columns"]["chat_room"][room]
 
         if self.ui_callback:
             self.ui_callback.remove_room(room)
 
-        self.core.pluginhandler.leave_chatroom_notification(room)
```

Comment:
Why was this moved?

Reply:
because a notification was raised for leaving a mistyped room that didn't exist. The triggering of the notification before the room is actually left would seem to defeat the purpose of having this notification.

Reply:
On second thought, I guess you get a join_chatroom_notification when reconnecting to the server, so sending one when closing a chatroom tab while disconnected doesn't make much sense.

> because a notification was raised for leaving a mistyped room that didn't exist.

We should bail out early so we don't trigger the UI callback.

Reply:
It is not really that clear whether `joined_rooms` is supposed to contain rooms that the user has selected to be in (i.e. open tabs), or if it is intended to be a list of active rooms that have been properly joined into by the server.

Since there seems to be some conflict about negotiating this between the client and server, perhaps this indicates the need for a seperate `selected_rooms` or `open_rooms` list that we always know reflects the open tabs in the client gui regardless of our ability or inability to establish a connection to the server for actually joining them.

That would help to ensure we don't accidentally leave any rooms by mistake that the user wanted to remain in, also help with allowing tabs to be closed while offline, and it may also potentially allow for us to build the ui widgets for displaying a status while waiting for the server to respond to the join request with the user list.